src/Microsoft.Health.Dicom.Api/Extensions/MediaTypeHeaderValueExtensions.cs

```diff
@@ -34,11 +34,18 @@ public static AcceptHeader ToAcceptHeader(this MediaTypeHeaderValue headerValue)
     {
         EnsureArg.IsNotNull(headerValue, nameof(headerValue));
         StringSegment mediaType = headerValue.MediaType;
+
         bool isMultipartRelated = StringSegment.Equals(KnownContentTypes.MultipartRelated, mediaType, StringComparison.OrdinalIgnoreCase);
+        // handle accept type with no quotes like "multipart/related; type=application/octet-stream; transfer-syntax=*"
+        bool? startsWithMultiPart = mediaType.Buffer?.StartsWith(KnownContentTypes.MultipartRelated, StringComparison.OrdinalIgnoreCase);
```

Comment:
The root of the problem is that the DICOMweb standard allows for the use of invalid content types. Should we instead attempt to fix those incoming content types so that downstream components don't need to worry? We could modify the header value with a regex that looks for this scenario, like:

```csharp
// As per RFC 2045, tokens may be any ASCII character except for space, control characters, and a set of special characters.
// [!#-:<-~]* - This is any printable ASCII character except for ' ', '"', and ';'
// [(),/:<=>?@[\\\\\\]] - This is the set of "special characters" that must be quoted, except for ';' and '"'
private static readonly Regex InvalidTypeParameterRegex = new Regex(
    "(\\s+|;)type=([!#-:<-~]*[(),/:<=>?@[\\\\\\]][!#-:<-~]*)( |;|$)",
    RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.CultureInvariant);

// ...

string validString = invalidTypeParameter.Replace(invalidContentType, x => $"{x.Groups[1]}type=\"{x.Groups[2]}\"{x.Groups[3]}");
```

Reply:
If there is a value add sure. Otherwise the hack is more readable and maintainable. We are letting ASP.Net to do best effort parsing and pining with a unit test if ASP.net changes